watch.go

```diff
@@ -70,11 +69,43 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			resp.Body.Close()
 		}()
 	scanning:
-		for scanner.Scan() {
+		for scanResult == nil {
 			// split event string
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
-			parts := strings.Split(scanner.Text(), "\n")
+
+			var evt []byte
+			var dat []byte
+			isPrefix := true
+			var result []byte
+
+			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
+			// we need bufio#ReadLine()
+			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
+			// so we have to add it to our result buffer.
+			evt, isPrefix, scanResult = scanner.ReadLine()
+			result = append(result, evt...)
+			result = append(result, '\n')
+
+			// 2. step: scan for the 'data:' part. Firebase returns just one 'data:'
+			// part, but the value can be very large. If we exceed a certain length
+			// isPrefix will be true until all data is read.
+			for {
```

Comment:
Might want to do

```go
for !isPrefix {
   dat, isPrefix, scanResult = scanner.ReadLine()
   result = append(result, dat...)
}
```

so that we don't call `scanner.ReadLine()` another time if `isPrefix` is already false from L#86